slurp/src/main/webapp/scripts/login.js

```diff
@@ -0,0 +1,24 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const firebaseUiConfig = {
+  signInSuccessUrl: 'view-trips.html',
```

Comment:
Does Firebase have its own page in the situation where sign in is unsuccessful?

Reply:
+1, as I don't know the answer to this either.

Reply:
On login error, FirebaseUI refreshes the login buttons to let the user try again. It _does_ have a callback function available that lets you handle the error yourself though. I have added this functionality, though all it does for now is print the error to console. I've opened up an Issue to display a generic error message to the user instead in a future PR.